network/src/connectivity_manager/mod.rs

```diff
@@ -596,6 +684,26 @@ fn log_dial_result(
     }
 }
 
+/////////////////////
+// DiscoverySource //
+/////////////////////
+
+impl DiscoverySource {
+    fn as_usize(self) -> usize {
```

Comment:
fwiw, I just use `discriminant`

Reply:
hmm, i'm not sure how to use discriminant to index into the array : d

Reply:
ah, I was using it to index a hashmap